types/stylelint/index.d.ts

```diff
@@ -50,3 +52,35 @@ export namespace formatters {
 }
 
 export function lint(options?: LinterOptions): Promise<LinterResult>;
+
+export namespace utils {
+    function report(violation: {
+        ruleName: string;
+        result: any;
+        message: string;
+        node: any;
+        index?: number;
+        word?: string;
+        line?: number;
+    }): void;
+
+    function ruleMessages(ruleName: string, messages: { [key: string]: any; }): typeof messages;
```

Comment:
For TS<=2.1 `typeof messages` would have to be changed to something like `{ [key: string]: any; }` or `object`